packages/tailwind-composer/src/types.ts

```
@@ -0,0 +1,411 @@
+  /**
+   * A method that allows the extension of the existing non-slotted component composer configuration.
+   * You are able to add new variants, add new values to existing variants, or even override existing variants.
+   * This method returns a new `NonSlottedComposerReturn` that includes the merged variants from the original
+   * configuration and the new configuration provided. This allows for a flexible way to build upon existing
+   * configurations without losing the original settings. This is particularly useful for creating reusable
+   * component configurations that can be extended in different parts of your application.
+   * @param newConfig - The new configuration to extend the existing non-slotted component composer.
+   * @returns The newly merged configuration
+   */
+  // eslint-disable-next-line @typescript-eslint/no-empty-object-type
+  extend: <TNewVariants extends NonSlottedVariants = {}>(newConfig?: {
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)